build_cmake/configs/modes.cmake

```diff
@@ -8,8 +8,56 @@
 
 # Establishes build configuration modes we can use when compiling.
 
+# Create an ASAN build variant
```

Comment:
These changes were a slight scope creep and can take it out of the PR if you rather it not be added.
Thought it might be useful to add some additional build modes to make it super easy to run the build+tests under ASan and UBSan (I personally always forget the flags).

Reply:
I'd definitely like to keep these. That's handy.